clients/src/main/java/org/apache/kafka/common/metrics/stats/Max.java

```diff
@@ -21,8 +21,16 @@
  */
 public final class Max extends SampledStat {
 
+    private final double initValue;
+
+    public Max(double initValue) {
+        super(initValue);
+        this.initValue = initValue;
```

Comment:
You should access `initialValue` from the superclass instead of duplicating it. And while you're at it, make `SampledStat.initialValue` and `SampledStat.samples` final.